File: src/main/java/kr/co/bookand/backend/bookstore/domain/BookStoreTheme.java
package kr.co.bookand.backend.bookstore.domain;

import lombok.*;

import javax.persistence.*;

@Entity
@Getter
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Builder
public class BookStoreTheme {
    @Id
    @GeneratedValue
    private Long id;

    @Enumerated(EnumType.STRING)
    private BookStoreType theme;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "bookstore_id")
    private BookStore bookStore;

    public void updateBookStore(BookStore bookStore) {
        this.bookStore = bookStore;
    }


}

File: src/main/kotlin/kr/co/bookand/backend/bookstore/domain/BookstoreTheme.kt
package kr.co.bookand.backend.bookstore.domain

import javax.persistence.*

@Entity
class BookstoreTheme(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "bookstore_theme_id")
    var id: Long = 0,

    @Enumerated(EnumType.STRING)
    var theme: BookstoreType,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "bookstore_id")
    var bookstore: Bookstore? = null
) {
    fun updateBookStore(bookstore: Bookstore) {
        this.bookstore = bookstore
    }
}